### ISSUE TITLE
Update mermaid to version 10

### ISSUE THREAD

**Issue by Ayowel** [gsantner/markor#2001]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **enhancement/feature request**. Not a [bug or question](https://github.com/gsantner/markor/issues/new/choose).
- [X] The topic is **not** already reported at [Issues](https://github.com/gsantner/markor/issues?q=#js-issues-search). _(I've searched it)_.
- [X] Markor **is** up to date. See [Releases](https://github.com/gsantner/markor/tags) for the latest version. Updates are available from [F-Droid](https://f-droid.org/en/packages/net.gsantner.markor/) and GitHub.
- [X] The wanted feature/enhancement is not present in the latest development version (git master). (Please [download](https://nightly.link/gsantner/markor/workflows/build-android-project/master) and try the test version of Markor, named **Marder**. Don't worry; Markor and Marder appear as completely separate applications. You can install both side-by-side, and Markor's settings are not touched. If your desired feature is present, you don't need to open this issue. The change will be part of the next Markor update.)

### Description

Hi,

Could the mermaid version be updated to v10.0.2 ? Later versions fix a few bugs and add timelines and mindmaps support as well as better git graphs support.
Considering that the file's weight is in the MB, it might be a good thing to handle it with git lfs. Unfortunately, GitHub does not allow to push files to LFS on forks, so I can't open a PR for this, but the changes are visible at https://github.com/Ayowel/markor-lfs .

If you want to reproduce on your own, here are the changes required after adding the files to git lfs (To add tracking : `git lfs track app/thirdparty/assets/mermaid/mermaid.min.js`, then stage the new `.gitattributes` and the file.):

```patch
diff --git a/.github/workflows/build-android-project.yml b/.github/workflows/build-android-project.yml
index 54ef541f..12afe5f0 100644
--- a/.github/workflows/build-android-project.yml
+++ b/.github/workflows/build-android-project.yml
@@ -22,12 +22,15 @@ jobs:
     steps:
 
     - name: "Checkout: Code"
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
+      with:
+        lfs: true
 
     - name: "Checkout: Code (PR)"
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
       if: "contains(github.event_name, 'pull_request')"
       with:
+        lfs: true
         ref: ${{github.event.pull_request.head.ref}}
         repository: ${{github.event.pull_request.head.repo.full_name}}
```

Unfortunately, mermaidjs does not provide a standalone release anymore, so the version must be built manually to allow offline use. I generated from an empty directory with the following files:

**webpack.config.js:**

```js
const path = require('path');

module.exports = {
  entry: './mermaid_exporter.js',
  mode: 'production',
  optimization: {
    minimize: true,
  },
  output: {
    asyncChunks: false,
    filename: 'mermaid.min.mjs',
    path: path.resolve(__dirname, 'dist'),
  },
};
```

**mermaid_exporter.js:**

```js
import * as mermaid from "mermaid";
for (var prop in mermaid) {
  exports[prop] = mermaid[prop];
}
```

**package.json:**

```json
{
  "scripts": {
    "build": "webpack-cli build -c webpack.config.js --progress"
  },
  "dependencies": {
    "mermaid": "^10.0.2",
    "webpack-cli": "^5.0.1"
  }
}
```

Then, simply run the following commands to generate the `mermaid.min.js` file in `dist`:

```sh
npm install
npm run build
```

### Information

Node version used for the mermaid build: v18.13.0
Built on Linux

### Source

Google Play

### Format / File type

Markdown

**Comment by gsantner** [comment]
Hello thanks for wanting to help here. The size of mermaid is about 1 Megabyte and lets say it's getting updated at Markor about once a year (introduction and first "1 mb addition" is from Feb2022).  While it might better fit to LFS, it makes other things more complex. If you have every week a 10 MB file to push of course the situation is different.

As you can see it makes things more complicated on CI (& F-Droid). Also for developers and contributors, because they have to manage and setup LFS separately and it's sometimes frustrating. That non-frequent changed file should not get a entrance barrier to the project. 

> Unfortunately, GitHub does not allow to push files to LFS on forks, so I can't open a PR for this,

You even self noting it, it even further may produce problems, as users might not be able to properly contribute as they would have by a simple normal PR.

So TL;DR: I prefer not to introduce and add LFS. But feel free to make a pull request which updates mermaid, in a normal git repository fashion without submodules/lfs/etc.

**Comment by gsantner** [comment]
Hello, good day @Ayowe 😄 

 Do you have still in mind to hellp with the update of mermaid?

**Comment by Ayowel** [comment]
I am, however I'm really uncomfortable with the idea of adding a non-lfs 3MB file that might be regularly updated as a dependency. Even more so considering that this is not an officially released file but a locally-built version without a verifiable checksum.

It looks like mermaidjs provides minified versions that should work offline again, I'll take a look next week and open a PR with mermaid 10.4.0 if it does.

**Comment by gsantner** [comment]
@Ayowel 
The thing is, it is not regularly updated. It doesn't always have to be the latest. For most people it should be enough to use the feature and making graphs etc - and that is supported in any version of it.
